lib/validation.rb

```diff
@@ -107,4 +110,47 @@ def self.validate_postgres_superuser_password(original_password, repeat_password
       fail ValidationFailed.new({"original_password" => messages.map { _1 }})
     end
   end
+
+  def self.validate_cidr(cidr)
+    NetAddr::IPv4Net.parse(cidr)
+  rescue NetAddr::ValidationError
+    fail ValidationFailed.new({cidr: "Invalid cidr"})
+  end
+
+  def self.validate_port_range(port_range)
+    if (match = port_range.match(ALLOWED_PORT_RANGE_PATTERN))
+      start_port = match[1].to_i
+
+      if match[2]
+        end_port = match[2].to_i
+        fail ValidationFailed.new({port_range: "Start port must be between 0 to 65535"}) unless (0..65535).cover?(start_port)
+        fail ValidationFailed.new({port_range: "End port must be between 0 to 65535"}) unless (0..65535).cover?(end_port)
+        fail ValidationFailed.new({port_range: "Start port must be smaller than end port"}) unless start_port <= end_port
+      else
+        fail ValidationFailed.new({port_range: "Port must be between 0 to 65535"}) unless (0..65535).cover?(start_port)
+      end
+    else
+      fail ValidationFailed.new({port_range: "Invalid port range"})
+    end
+  end
+
+  def self.validate_request_body(request_body, required_keys, allowed_optional_keys = nil)
+    begin
+      request_body_params = JSON.parse(request_body)
```

Comment:
Doesn't [json_parser](https://roda.jeremyevans.net/rdoc/classes/Roda/RodaPlugins/JsonParser.html) do this check already? We might pass `r.params` directly to it.

Reply:
I found the reason. It expects to have `Content-Type: application/json` header to parse it as JSON. The client might not put that. Let's keep your version.